src/pages/Swap.tsx

```diff
@@ -248,11 +250,14 @@ function Swap(): ReactElement {
         formStateArg.swapType === SWAP_TYPES.TOKEN_TO_TOKEN &&
         bridgeContract != null
       ) {
-        const originPool = POOLS_MAP[formStateArg.from.poolName]
-        const destinationPool = POOLS_MAP[formStateArg.to.poolName]
+        const originPool = basicPools?.[formStateArg.from.poolName]
+        const destinationPool = basicPools?.[formStateArg.to.poolName]
         const [amountOutSynth, amountOutToken] =
           await bridgeContract.calcTokenToToken(
-            [originPool.addresses[chainId], destinationPool.addresses[chainId]],
+            [
+              originPool?.metaSwapDepositAddress ?? "",
```

Comment:
this will fail if we don't know the address. Write to the `error` var and set `amountToReceive, amountMediumSynth` to `Zero` instead